homeassistant/helpers/config_validation.py

```diff
@@ -233,8 +233,14 @@ def template(value):
     """Validate a jinja2 template."""
     if value is None:
         raise vol.Invalid('template value is None')
-    if isinstance(value, (list, dict)):
-        raise vol.Invalid('template value should be a string')
+    if isinstance(value, list):
```

Comment:
Thinking about this again I dont think we should modify the validator. 
What if you really just want a string for `cv.template`? Isn't all current uses like this?

Cant you get template validation with a simple string function by using:

```python
vol.Schema({
  'data_template_string': cv.template,
  'data_template_array': [cv.template],
  'data_template_dict': {cv.string: cv.template}
})

That forces you to build your schema correctly